src/Core/src/Signer/SignerV4.php

```diff
@@ -59,15 +59,25 @@ public function __construct(string $scopeName, string $region)
 
     public function presign(Request $request, Credentials $credentials, RequestContext $context): void
     {
-        $now = $context->getCurrentDate() ?? new \DateTimeImmutable();
-        $expires = $context->getExpirationDate() ?? (new \DateTimeImmutable($now->format(\DateTimeInterface::ATOM)))->add(new \DateInterval('PT1H'));
+        if (null === $now = $context->getCurrentDate()) {
+            $now = new \DateTimeImmutable();
+        } else {
+            $now = new \DateTimeImmutable($now->format(\DateTimeInterface::ATOM));
+        }
+        $now = $now->setTimezone(new \DateTimeZone('GMT'));
```

Comment:
why `GMT` rather than `UTC` (which is what `Z` as timezone indicator means) ?

Reply:
dummy copy/paste from official AWS SDK that use GMT.

Indeed, the doc refer to UTC https://docs.aws.amazon.com/general/latest/gr/sigv4-date-handling.html.

Fixed